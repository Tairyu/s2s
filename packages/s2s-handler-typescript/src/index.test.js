// @flow
import handler from '.'
import plugin from './__tests__/helpers/plugin'

const setup = (plug = plugin) => {
  return {
    eventPath: '/path/to/fixture/code.js',
    filename: 'dummy.js',
    plugin: { test: /dummy/, plugin: plug },
  }
}

const code = `
function greeter(person: string) {
  return "Hello, " + person;
}

let user = "Jane User";
greeter(user);
`

test('ts-handlerは変換後のコードを返す', () => {
  const result = handler(code, setup())
  expect(result).toMatchSnapshot()
})

test('pluginがArrayの場合', () => {
  const result = handler(code, setup([plugin, { x: 1 }]))
  expect(result).toMatchSnapshot()
})

test('codeが""の場合、""が返ること', () => {
  const result = handler('', setup())
  expect(result).toBe('')
})
