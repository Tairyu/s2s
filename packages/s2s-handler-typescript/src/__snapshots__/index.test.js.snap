// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pluginがArrayの場合 1`] = `
"function greeterfixture(personfixture: string) {
  return \\"Hello, \\" + personfixture;
}

let userfixture = \\"Jane User\\";
greeterfixture(userfixture);"
`;

exports[`ts-handlerは変換後のコードを返す 1`] = `
"function greeterfixture(personfixture: string) {
  return \\"Hello, \\" + personfixture;
}

let userfixture = \\"Jane User\\";
greeterfixture(userfixture);"
`;
